app/models/package_manager/base.rb

```diff
@@ -92,25 +92,43 @@ def self.map_project(project)
         &.delete_if { |_key, value| value.blank? }
     end
 
-    def self.update_version(name, version)
-      unless self::SUPPORTS_SINGLE_VERSION_UPDATE
-        logger.warn("#{db_platform}.update_version(#{name}, #{version}) called but not supported on platform")
+    def self.update(name, sync_version: :all)
+      if sync_version != :all && !self::SUPPORTS_SINGLE_VERSION_UPDATE
+        logger.warn("#{db_platform}.update(#{name}, sync_version: #{sync_version}) called but not supported on platform")
         return
       end
 
-      update(name, sync_versions: false)
       raw_project = project(name)
+      return false unless raw_project.present?
```

Comment:
I'm looking at this all a bit cross-eyed and it feels like it could be broken down into smaller functions, but combining seems like a good step in the right direction before refactoring/extracting.